substra/sdk/client.py

```diff
@@ -291,6 +291,7 @@ def add_dataset(self, data, exist_ok=False):
                 "public": bool,
                 "authorized_ids": list[str],
             },
+            "metadata": list[str]
```

Comment:
`list[str]` means that acceptable values are like `['a', 'b', 'c']`. Instead we want acceptable values to be `{'a': 'b', 'c': 'd'}`.

In python typing, this is a Dict and not a List : https://docs.python.org/3/library/typing.html#typing.Dict

Reply:
Yes, you're right!